filetracker/client/lock_manager.py

```diff
@@ -64,8 +69,13 @@ def lock_exclusive(self):
         def unlock(self):
             fcntl.flock(self.fd, fcntl.LOCK_UN)
 
-        def close(self):
+        def close(self, delete=False):
             if self.fd != -1:
+                if delete:
+                    os.remove(self.filename)
+                    dir_path = os.path.dirname(self.filename)
+                    if dir_path != self.manager.dir:
+                        os.removedirs(dir_path)
```

Comment:
Czy ten if jest po to, żeby nie usunąć katalogu `.filetracker-store/locks` ?
Bo jeśli tak, to to chyba nie działa.
Jesli `dir_path == ".filetracker-store/locks/foo/bar/"` uruchomi się na tej ścieżce `os.removedirs` i usunie również katalog `.filetracker-store/locks` jeśli będzie on pusty.

Reply:
Ten kod jest wzorowany na tym:
https://github.com/sio2project/filetracker/blob/master/filetracker/client/local_data_store.py#L65

Brakuje jednakże obsługi błędu, kiedy to usuwanie się wysypie z powodu wyścigu lub niepustości dir_path (os.removedirs rzuca wyjątkiem, jeśli nie można skasować wskazanego katalogu).

Spróbuję to zrobić jednak pewniej jeśli chodzi o odporność na wyścigi z usuwaniem i tworzeniem tych folderów.

Reply:
Może w pierwszym przybliżeniu wystarczy usunąc 1 poziom folderów, tzn. zwykłe rmdir?

Reply:
To też jest jakieś rozwiązanie, jeśli to co napisałem w międzyczasie nie wygląda dobrze.